pkgs/data/themes/graphite/default.nix

```diff
@@ -0,0 +1,91 @@
+{ lib
+, stdenvNoCC
+, fetchFromGitHub
+, gnome-themes-extra
+, gtk-engine-murrine
+, jdupes
+, sassc
+, themeVariants ? [] # default: blue
+, colorVariants ? [] # default: all
+, sizeVariants ? [] # default: standard
+, tweaks ? []
+, wallpapers ? false
+}:
+
+let
+  assertSubListOf = name: val: xs:
+    let
+      unexpected = lib.subtractLists xs val;
+    in
+      lib.assertMsg (unexpected == [])
+        "${name} must be a sublist of ${
+          lib.generators.toPretty {} xs}, but contains: ${
+            lib.generators.toPretty {} unexpected}";
+
+in
+assert assertSubListOf "theme variants" themeVariants [ "default" "purple" "pink" "red" "orange" "yellow" "green" "teal" "blue" "all" ];
+assert assertSubListOf "color variants" colorVariants [ "standard" "light" "dark" ];
+assert assertSubListOf "size variants" sizeVariants [ "standard" "compact" ];
+assert assertSubListOf "tweaks" tweaks [ "nord" "black" "midblack" "rimless" "normal" ];
+
+stdenvNoCC.mkDerivation rec {
+  pname = "graphite-gtk-theme";
+  version = "unstable-2022-01-04";
+
+  src = fetchFromGitHub {
+    owner = "vinceliuice";
+    repo = pname;
+    rev = "947cac4966377d8f5b5a4e2966ec2b9a6041d205";
+    sha256 = "11pl8hzk4fwniqdib0ffvjilpspr1n5pg1gw39kal13wxh4sdg28";
+  };
+
+  nativeBuildInputs = [
+    jdupes
+    sassc
+  ];
+
+  buildInputs = [
+    gnome-themes-extra
+  ];
+
+  propagatedUserEnvPkgs = [
+    gtk-engine-murrine
+  ];
+
+  # These fixup steps are slow and unnecessary.
+  dontPatchELF = true;
+  dontRewriteSymlinks = true;
```

Comment:
Are those really that slow? I also think setting `dontPatchElf` is the reason for the manual need of `patchShebangs` no :thinking: ?

Reply:
The packaged content is not ELF binaries, and `jdupes` already takes care of linking. This was inspired by https://github.com/NixOS/nixpkgs/pull/121534#commits-pushed-6f55db1

In this package, `patchShebangs` is needed to fix a build script, and not a installed one, and because of that it has to be explicit in the nix expression. Shebangs are fixed automatically in the `fixupPhase` for installed files only, as far as I know.

Reply:
But for this package they are not really that slow. Therefore I will remove this settings.

Reply:
I am still not sure if those optimizations are really worth. Would be better to either fix this at the `stdenv` level (so all packages could have the same optimizations) or at least some documentation about this, otherwise the gains are very minimal.

Reply:
But anyway, fair enough.

Reply:
Done.